components/remote_settings/src/storage.rs

```diff
@@ -60,39 +101,373 @@ impl Storage {
     /// passed to `set_attachment`.
     pub fn get_attachment(
         &self,
-        _collection_url: &str,
-        _attachment_id: &str,
+        collection_url: &str,
+        attachment_id: &str,
     ) -> Result<Option<Attachment>> {
-        Ok(None)
+        let conn = self.conn.lock().unwrap();
+        let mut stmt =
+            conn.prepare("SELECT data FROM attachments WHERE id = ? AND collection_url = ?")?;
+        let result: Option<Vec<u8>> = stmt
+            .query_row(params![attachment_id, collection_url], |row| row.get(0))
+            .optional()?;
+        if let Some(data) = result {
+            let attachment: Attachment = serde_json::from_slice(&data)?;
+            Ok(Some(attachment))
+        } else {
+            Ok(None)
+        }
     }
 
     /// Set the list of records stored in the database, clearing out any previously stored records
     pub fn set_records(
         &self,
-        _collection_url: &str,
+        collection_url: &str,
         records: &[RemoteSettingsRecord],
     ) -> Result<()> {
-        for record in records {
-            println!("Should store record: {record:?}");
+        let mut conn = self.conn.lock().unwrap();
+
+        let tx = conn.transaction()?;
+
+        // Delete existing records for this collection_url
+        tx.execute(
+            "DELETE FROM records WHERE collection_url = ?",
+            params![collection_url],
+        )?;
+
+        // Insert new records
+        {
+            let mut stmt =
+                tx.prepare("INSERT INTO records (id, collection_url, data) VALUES (?, ?, ?)")?;
+            for record in records {
+                let data = serde_json::to_vec(record)?;
+                stmt.execute(params![record.id, collection_url, data])?;
```

Comment:
Nit: I believe you could pass a tuple for this.  The params! macro is only needed for weird cases.

Reply:
Ahh, I'm pretty sure you're right about that.